Лабораторная работа №1

ippi12 opened the pull request:
Сделал окно входа и окно регистрации, сейчас разбираюсь с валидацией данных

ippi12 committed:
Add files via upload

dkorkin commented on Authentification/Authentification/AppDelegate.swift:
```diff
@@ -0,0 +1,25 @@
+    var window: UIWindow?
+
+    func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
+        let viewController = ViewController()
```
нейминг AuthViewController ?

dkorkin commented on Authentification/AuthentificationTests/AuthentificationTests.swift:
```diff
@@ -0,0 +1,33 @@
+//  Created by Компьютер on 07.02.2022.
+//
+
+import XCTest
```
не используешь тесты, убирай файл. и таргет тестов

dkorkin commented on Authentification/AuthentificationUITests/AuthentificationUITestsLaunchTests.swift:
```diff
@@ -0,0 +1,32 @@
+        // such as logging into a test account or navigating somewhere in the app
+
+        let attachment = XCTAttachment(screenshot: app.screenshot())
+        attachment.name = "Launch Screen"
```
ну ладно)

dkorkin commented on Authentification/Cells/CreatePasswordTableViewCell.swift:
```diff
@@ -0,0 +1,61 @@
+    
+    func setupUI() {
+        selectionStyle = .none
+        //loginView.translatesAutoresizingMaskIntoConstraints = false
```
лишние комменты убираем

dkorkin commented on Authentification/Cells/EmailTableViewCell.swift:
```diff
@@ -0,0 +1,60 @@
+
+import UIKit
+
+class EmailTableViewCell: UITableViewCell {
```
CreatePasswordTableViewCell аналогична этой, значит ее проще конфигурировать через проперти интового енама authForType { .email = 0 ; password = 1} и тд

dkorkin commented on Authentification/ViewControllers/RegistrationViewController.swift:
```diff
@@ -0,0 +1,147 @@
+    let signButton: UIButton = {
+        let button = UIButton()
+        button.backgroundColor = .systemBlue
+        button.setTitle("Регистрация", for: .normal)
```
локализуй  и саму кнопку добавить во view и tableFooterView или tableHeaderView

dkorkin commented on Authentification/ViewControllers/RegistrationViewController.swift:
```diff
@@ -0,0 +1,147 @@
+        myTableView.separatorStyle = .none
+        myTableView.dataSource = self
+        myTableView.delegate = self
+        }
```
лишняя табуляция

dkorkin commented on Authentification/ViewControllers/RegistrationViewController.swift:
```diff
@@ -0,0 +1,147 @@
+        NSLayoutConstraint.activate([
+            signButton.topAnchor.constraint(equalTo: myTableView.bottomAnchor, constant: 0),
+//            signButton.bottomAnchor.constraint(equalTo: myTableView.bottomAnchor, constant: 100),
+//            signButton.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: 100),
```
комменты

dkorkin commented on Authentification/ViewControllers/ViewController.swift:
```diff
@@ -0,0 +1,146 @@
+
+import UIKit
+
+final class ViewController: UIViewController, UITableViewDelegate, UITableViewDataSource {
```
по факту это разные датасорсы таблицы 1 и той же

ippi12 committed:
Add files via upload

Я удалил лишние комментарии, удалил лишние ячейки, добавил segmentedcontrol в Header, button в Footer
